docs/src/high_dimension.rst

```diff
@@ -4,3 +4,145 @@
 ===========================
+
+Naive inference in high dimension is ill-posed
+----------------------------------------------
+
+In some cases, data represent high-dimensional measurements of some phenomenon of interest (e.g. imaging or genotyping). The common characteristic of these problems is to be very high-dimensional and lead to correlated features. Both aspects are clearly detrimental to conditional inference, making it both expensive and powerless:
+
+* Expensive: most learers are quadratic or cubic in the number of features. Moreover per-feature inference generally entails a loop over features
+* powerless: As dimensionality and correlation increase, it  becomes harder and harder to isolate the contribution of each variable, meaning that conditional inference is ill-posed.
+
+This is illustrated in the above example, where the Desparsified Lasso struggles
+to identify relevant features. We need some data to start::
+
+    n_samples = 100
+    shape = (40, 40)
+    n_features = shape[1] * shape[0]
+    roi_size = 4  # size of the edge of the four predictive regions
+
+    # generating the data
+    from hidimstat._utils.scenario import multivariate_simulation_spatial
+    X_init, y, beta, epsilon = multivariate_simulation_spatial(
+        n_samples, shape, roi_size, signal_noise_ratio=10., smooth_X=1
+    )
+
+Then we perform inference on this data using the Desparsified Lasso::
+
+    from hidimstat.desparsified_lasso import (
+    desparsified_lasso,
+    desparsified_lasso_pvalue,
+    )
+    beta_hat, sigma_hat, precision_diagonal = desparsified_lasso(X_init, y)
+    _, pval_corr, _, one_minus_pval_corr, cb_min, cb_max = (
+        desparsified_lasso_pvalue(X_init.shape[0], beta_hat, sigma_hat, precision_diagonal)
+    )
+        
+    # compute estimated support (second method)
+    import numpy as np
+    alpha = .05
+    selected_dl = np.logical_or(pval_corr < alpha , one_minus_pval_corr < alpha)
+    print(f'Desparsified Lasso selected {np.sum(selected_dl)} features among {np.sum(beta > 0)} ')
+
+
+
+Feature Grouping and its shortcomings
+-------------------------------------
+
+As discussed earlier, feature grouping is a meaningful solution to deal with such cases: it reduces the number of features to condition on, and generally also decreases the level of correlation between features.
+
+.. seealso::
+
+   * The :ref:`Grouping documentation <grouping>`
+
+
+As hinted in :footcite:t:`meinshausen2009pvalues` an efficient way to deal with such configuration is to take the per-group average of the features: this leads to a *reduced design*. After inference, all the feature in a given group obtain the p-value of the group representative. When the inference engine is Desparsified Lasso, the resulting method is called Clustered Desparsified lasso, or **CluDL**.
+
+Using the same example as previously, we start by defining a clustering method that will perform the grouping. For image data, Ward clustering is a good default model::
```

Comment:
Should we add a bit more details explaining that the clustering is constrained by pixel's neighborhood?